c2rust-transpile/tests/snapshots/snapshots__transpile@macros.c.snap

```diff
@@ -29,287 +29,245 @@ pub struct fn_ptrs {
     pub fn2: Option<unsafe extern "C" fn(std::ffi::c_int) -> std::ffi::c_int>,
 }
 pub type zstd_platform_dependent_type = std::ffi::c_long;
-pub const NESTED_INT: std::ffi::c_int = 0xffff as std::ffi::c_int;
+pub const true_0: std::ffi::c_int = unsafe { 1 as std::ffi::c_int };
```

Comment:
@fw-immunant, note that this does achieve a lot of what #290 does.  It's similar to the portable types, stuff.  There, we'll now emit a `uint32_t` type alias instead of a `u32` directly, and this will emit a `true_0` `const` instead of a `true` literal directly.  Both could then be refactored pretty easily.  I'm not exactly sure how #290 works, but the advantage here is that it matches where the C used the `true` and `false` macros, and if C uses 0 and non-zero directly, then it keeps that behavior.  And then I'm not sure how #290 handles `int`s vs `_Bool`s either.

We can also resurrect #295 to use `r#true` raw identifiers instead if that's helpful.

Reply:
There definitely is some overlap in spirit but I think it's probably worthwhile to address this from both sides--this PR will clean up some uses of `#define` symbols, but it seems like we're still mostly translating them as integers.

Do you know what might help us translate `LITERAL_BOOL` as `const LITERAL_BOOL: bool = ...;`? It seems like `recreate_const_macro_from_expansions` might be able to realize that all uses are boolifying it.

Reply:
> Do you know what might help us translate `LITERAL_BOOL` as `const LITERAL_BOOL: bool = ...;`? It seems like `recreate_const_macro_from_expansions` might be able to realize that all uses are boolifying it.

I'm not sure, but I'll look into it.